### PR TITLE
hotfix smp crash.

### LINKED ISSUE #378: minecolonies 1.11 beta 6 build 2103 crashes dedicated Forge 1.11 build 2199 while building hut
crash report:
https://gist.github.com/5e212cdde069c2cb5f2c77748ee2fd18

fml server latest:
https://gist.github.com/fd0806679881afc0ecbc29b12c17a10e

version 1.11-0.6.2103

### Expected behavior

started town hall on water
set builder hut on shore
builder tries to build hut on water

also builder doesn't actually say anything in chat as to what he needs
just things like item.blahblah.toolNeeded <-- not literally but darn close to it ... etc  ... but never the tool name or the material name

### Actual behaviour

builder bobs around in the water
server crashes

### Steps to reproduce the problem

1. place builder chest on shore facing water with about 30x30 of water in front of it
2. wait for builder to crash server

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #378 

# Changes proposed in this pull request:
- Fix the sideOnly problem which crashes the server.

Review please

### CODE REVIEW

**Comment by Raycoms** [comment]
@Minecolonies/maintainers

File: src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java [hunk 1]
```diff
@@ -471,40 +486,40 @@ public void loadStructure(@NotNull final String name, int rotateTimes, BlockPos
...
     }
 
+    /**
+     * Iterates through all the required resources and stores them in the building.
+     */
     private void requestMaterials()
     {
-        //We need to deal with materials
-        if (!Configurations.builderInfiniteResources && job instanceof JobBuilder)
+        JobBuilder builderJob = (JobBuilder) job;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java]: MINOR Local variable 'builderJob' could be declared final [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 1 issue
* MINOR 1 minor

Watch the comments in this conversation to review them.

**Comment by marvin-bitterlich** [comment]
LGTM

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/380/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Pull request merged**